### PR TITLE
Clean up generator track1 tool code

### PR DESCRIPTION
**Pull request by Alancere**
https://github.com/Azure/azure-sdk-for-go/issues/23155

### CODE REVIEW

**Review by tadelesh — COMMENTED** [review]
it is possible to consolidate all the remaining part of `autorest`/`model`/... to `release/v2` and rearrange all the tool strucutre?

**Comment by Alancere** [comment]
> it is possible to consolidate all the remaining part of `autorest`/`model`/... to `release/v2` and rearrange all the tool strucutre?

updated, please review.

File: eng/tools/generator/config/options.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-package model
+package config
```
Comment (raych1) [inline eng/tools/generator/config/options.go:4]: What are the differences between `option.go` and `options.go`? Can we merge them?

Comment (Alancere) [reply eng/tools/generator/config/options.go]: merged options.go in option.go

**Review by tadelesh — APPROVED** [review]
please update the readme file

**Comment by tadelesh** [comment]
@benbp could you help to approve this pr?

**Review by weshaggard — APPROVED** [review]

**Comment by weshaggard** [comment]
I've approved but for future stuff we should update the CODEOWNERS file to add the owners for this tool to help with the new codeowner approval requirement.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -103,6 +103,9 @@
 # PRLabel: %EngSys
 /sdk/template/                @benbp @weshaggard
 
+# Add owners for go sdk tool
```
Comment (RickWinter) [inline .github/CODEOWNERS:106]: Please move this up be after the /eng/common/pipeline line

**Review by RickWinter — APPROVED** [review]

**Pull request merged**